src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Templates/TemplateInserterStep.cs

```diff
@@ -71,7 +71,7 @@ public TemplateInserterStep(TemplateProvider templateProvider, ILogger<TemplateI
             }
         }
 
-        protected override bool IsApplicableImpl(IUpgradeContext context) => context?.CurrentProject is not null && _templateProvider.TemplateConfigFileNames.Any();
+        protected override bool IsApplicableImpl(IUpgradeContext context) => context?.CurrentProject is not null && context.CurrentProject.Language == Languages.CSharp && _templateProvider.TemplateConfigFileNames.Any();
```

Comment:
I don't think this belongs in IsApplicableImpl. This scenario should never come up thanks to the ready check but if a user somehow does manage to try upgrading a WPF VB app, we probably want to fail rather than silently skip this step.

Reply:
The TemplateInserterStep.cs adds `Startup.cs` and `Program.cs` to web apps. If the project is not `CSharp` then we don't want CSharp files.

Reply:
Good point. I totally overlooked that this was the TemplateInserterStep (I thought we were still updating the TryConvertUpgradeStep). As we discussed offline, the best thing to do will be to fix the template inserter step to work with multiple languages.